python_modules/libraries/dagster-pyspark/dagster_pyspark/utils.py

```diff
@@ -4,18 +4,20 @@
 import dagster._check as check
 
 
-def build_pyspark_zip(zip_file, path):
+def build_pyspark_zip(zip_file, path, skip_hidden=True, follow_links=False):
     """Archives the current path into a file named `zip_file`."""
     check.str_param(zip_file, "zip_file")
     check.str_param(path, "path")
 
     with zipfile.ZipFile(zip_file, "w", zipfile.ZIP_DEFLATED) as zf:
-        for root, _, files in os.walk(path):
+        for root, _, files in os.walk(path, followlinks=follow_links):
             for fname in files:
                 abs_fname = os.path.join(root, fname)
 
                 # Skip various artifacts
                 if "pytest" in abs_fname or "__pycache__" in abs_fname or "pyc" in abs_fname:
```

Comment:
We could improve this to support a set of regular expressions:

- then we can pass custom exclusions
- we can improve the current code to exclude `\.pyc$` instead of any filename with `pyc` in it